src/library_sdl.js

```diff
@@ -2860,9 +2861,12 @@ var LibrarySDL = {
       audio.frequency = info.audio.frequency;
     }
     audio['onended'] = function() { // TODO: cache these
-      if (channelInfo.audio == this) { channelInfo.audio.paused = true; channelInfo.audio = null; }
+      if (channelInfo.audio === this || channelInfo.audio.webAudioNode === this) { channelInfo.audio.paused = true; channelInfo.audio = null; }
```

Comment:
Can you  split the body over multiple lines here?

Reply:
Do you think its a bug there `this` can be `channelInfo.audio` in once case but `channelInfo.audio.webAudioNode` in another case?   Do you know why we have both of those modes?